src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -67,38 +83,65 @@ class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
       await logger.logToStderr(`Restoring items from recycle bin at ${args.options.siteUrl}...`);
     }
 
-    const requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin/RestoreByIds`;
+    let requestUrl: string = `${args.options.siteUrl}/_api`;
 
-    const ids: string[] = formatting.splitAndTrim(args.options.ids);
-    const idsChunks: string[][] = [];
+    try {
+      if (args.options.ids) {
+        requestUrl += `/site/RecycleBin/RestoreByIds`;
+        const ids: string[] = formatting.splitAndTrim(args.options.ids);
+        const idsChunks: string[][] = [];
 
-    while (ids.length) {
-      idsChunks.push(ids.splice(0, 20));
-    }
+        while (ids.length) {
+          idsChunks.push(ids.splice(0, 20));
+        }
 
-    try {
-      await Promise.all(
-        idsChunks.map((idsChunk: string[]) => {
-          const requestOptions: any = {
-            url: requestUrl,
-            headers: {
-              'accept': 'application/json;odata=nometadata',
-              'content-type': 'application/json'
-            },
-            responseType: 'json',
-            data: {
-              ids: idsChunk
-            }
-          };
-
-          return request.post(requestOptions);
-        })
-      );
+        await Promise.all(
+          idsChunks.map((idsChunk: string[]) => {
+            const requestOptions: CliRequestOptions = {
+              url: requestUrl,
+              headers: {
+                'accept': 'application/json;odata=nometadata',
+                'content-type': 'application/json'
+              },
+              responseType: 'json',
+              data: {
+                ids: idsChunk
+              }
+            };
+
+            return request.post(requestOptions);
+          })
+        );
+      }
+      else {
+        if (args.options.allPrimary) {
+          requestUrl += `/web/RecycleBin/RestoreAll`;
+          await this.restoreRecycleBinItems(requestUrl);
```

Comment:
When specifying an API URL, let's use this method to unify it a bit more. Let's do this on the other places as well.

```suggestion
          await this.restoreRecycleBinItems(baseUrl + '/web/RecycleBin/RestoreAll');
```